### PR TITLE
feat: implemention `TagService`

### PR DESCRIPTION
**Pull request by Alex8118**
Create Tag with validation

### CODE REVIEW

File: src/Services/Identity/Idn.Plugin/TagService.cs [hunk 1]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (denmaklucky) [inline src/Services/Identity/Idn.Plugin/TagService.cs]: You can delete the empty file

File: src/Services/Tag.Contracts/Tag.Contracts.csproj [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
```
Comment (denmaklucky) [inline src/Services/Tag.Contracts/Tag.Contracts.csproj]: not needed. Check all placess as well.

File: src/Services/Tag.Contracts/Tag.DataAccess/Tag.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Tag.Contracts;
+
+namespace Tag.DataAccess
```
Comment (denmaklucky) [inline src/Services/Tag.Contracts/Tag.DataAccess/Tag.cs]: 1 - check the spaces (the formating a little weird)
2 - the constructor should come frist and then properties
3 - use file scoped namespace 
4 - the properties should not have `set`
5 - between properties should be an one line

File: src/Services/Tag.Contracts/Tag.Extensions/TagService.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace Tag.Extensions;
+
+	public class TagService
```
Comment (denmaklucky) [inline src/Services/Tag.Contracts/Tag.Extensions/TagService.cs]: 1 - check formating, please
2 - don't need this file :)

File: src/Services/Tag.Contracts/TagRequest.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using MediatR;
+using Tag.DataAccess.Tag;
+using Tag.Extensions;
+
+namespace Tag.Contracts;
+
+public sealed record TagRequest(Tag tag, Source Source) : IRequest<TagResponse>;
+
```
Comment (denmaklucky) [inline src/Services/Tag.Contracts/TagRequest.cs]: remove one line, please

File: src/Services/Tag/Tag.DataAccess/TagRepository.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Tag.DataAccess;
+
+public sealed class TagRepository
```
Comment (denmaklucky) [inline src/Services/Tag/Tag.DataAccess/TagRepository.cs]: check out the current strcuture of the folders into service and modify yours :)

File: src/Services/Tag/Tag.Domain/ServiceCollectionExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+using TagContracts;
+
+namespace Tag.Domain;
+
+public static class ServiceCollectionExtensions
+{
+    public static IServiceCollection AddTags(this IServiceCollection services)
+    {
+        services.AddSingleton<IValidator<CreateTagRequest>, CreateTagRequestValidator>();
+        return services;
```
Comment (denmaklucky) [inline src/Services/Tag/Tag.Domain/ServiceCollectionExtensions.cs]:
```suggestion
        return services
               .AddSingleton<IValidator<CreateTagRequest>, CreateTagRequestValidator>();
```

File: src/Services/Tag/Tag.Plugin/ServiceTagExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+public static class ServiceTagExtensions
+{
+    
+    public static IServiceCollection AddTagService(this IServiceCollection services)
+    {
+        return services
+            .AddMediatR(typeof(MediatorEntryPoint).Assembly)
+            .AddTags();
+    }
+
```
Comment (denmaklucky) [inline src/Services/Tag/Tag.Plugin/ServiceTagExtensions.cs]: empty line :)

File: src/WebApi/Endpoints/TagsEndpoints.cs [hunk 9]
```diff
@@ -1,19 +1,36 @@
...
-    }
+        private const string BaseRoute = "api/v1/tags";
 
-    private static async Task<IResult> CreateAsync([FromServices] IUserIdAccessor userIdAccessor)
-    {
-        return Results.Ok(userIdAccessor.UserId);
+        public static void MapTagsEndpoints(this WebApplication application)
+        {
+            application.MapPut($"{BaseRoute}", CreateTagAsync);
+            application.MapPut($"{BaseRoute}", GetTagsAsync);
```
Comment (denmaklucky) [inline src/WebApi/Endpoints/TagsEndpoints.cs:15]: not sure that PUT should works as GET http verb :)
Change to `MapGet`, please

File: src/WebApi/Endpoints/TagsEndpoints.cs [hunk 10]
```diff
@@ -1,19 +1,36 @@
...
+            GetTagRequest request, 
+            [FromServices] IMediator mediator)
+        {
+            var response = await mediator.Send(request);
+            return Results.Ok(response);
+        }
     }
-}
\ No newline at end of file
+}
```
Comment (denmaklucky) [inline src/WebApi/Endpoints/TagsEndpoints.cs:36]: not need spaces :)

File: src/Services/Tag.Contracts/Tag.DataAccess/Tag.cs (hunk 3 above)
Comment (Alex8118) [reply src/Services/Tag.Contracts/Tag.DataAccess/Tag.cs:4]: Tag.Contracts.csproj deleted

**Pull request merged**